### ISSUE TITLE
Shows Unavailable when blocking Internet & DNS

### ISSUE THREAD

**Issue by Nuuki9** [rospogrigio/localtuya#1480]
I'm struggling with getting a fully localised setup working. I've gone through discussions here and thought I understood what was needed, but no luck. I'm using a UniFi Dream Machine Pro SE.

I have a number of bulbs that are provisioned and working - I can control them using LT just fine, when they have Internet access. They sit in a dedicated VLAN (separate VLAN to Home Assistant) and I've now configured the firewall to limit outbound connections as follows:

- Allow access to Home Assistant server (all ports).
- Block access to all other local networks.
- Block all Internet access.
- Block all DNS requests (to router or other remote hosts).

This works initially, but if I reboot a device, it shows an Unavailable in Home Assistant. I hoped that blocking DNS would resolve this, but no luck. Once I open up Internet access again, it works (even with DNS still blocked). I have uninstalled the Smart Life app from all devices.

So, am I misunderstanding what I need to filter, or do I have the right understanding? If this _should_ work then presumably I'm implementing the firewall rules wrong. I'd be fine with that, but wanted to check I had the right undertanding.

**Comment by xZetsubou** [comment]
There were an issue before but I don't know if your case is the same, If you want you can try installing [my fork](https://github.com/xZetsubou/hass-localtuya) and test it.
Whether it's fixed or not let me please.

**Comment by Nuuki9** [comment]
@xZetsubou I'll give that a try.

**Comment by feene** [comment]
> @xZetsubou I'll give that a try.

What did you give a try?  I am in the same boat.  Blocking access to DNS servers in my home vlan and internet.  Tuya Devices go unavailable.

**Comment by Nuuki9** [comment]
That's odd - pretty sure there was a response that I replied to. No matter - I haven't yet been able to resolve it. Having gone through what I felt were fairly systematic checks, I wasn't able to spend more time on it at that point. I guess my hope is that when I next get around to looking at it, that someone else will have provided some clarity on what's needed.

Certainly if you dig into this and want to compare notes, I'm happy to do that.

**Comment by lankhaar** [comment]
This is an older post, but in case you still haven't got this to work I'll share an important details here, I hope it help.

Make sure that you don't block ALL incoming/outgoing traffick from/to the devices, only the traffick from/to ethernet. If you'd block all traffick, this would also block the requests done over your LAN (which home assistant does to communicate to your tuya devices).

**Comment by Nuuki9** [comment]
Thanks for the comment @lankhaar. Indeed I did not get this resolved, and it went into my homelab backlog. As I recall, these are the rules I applied:

- Block *all* outbound DNS requests (including to local resolvers).
- Allow outbound connections to Home Assistant server only.
- Allow all inbound connections.

So to your point, I was indeed allowing connections to HA, but no luck. Ubiquiti have added some new features for configuring rules so I may take another look in due course. Right now I only have a coupld of Tuya bulbs so I may simply reaplce those in time, but its still annoying not to have solved this.

**Comment by snzsnz** [comment]
> Thanks for the comment [@lankhaar](https://github.com/lankhaar). Indeed I did not get this resolved, and it went into my homelab backlog. As I recall, these are the rules I applied:
> 
> * Block _all_ outbound DNS requests (including to local resolvers).
> * Allow outbound connections to Home Assistant server only.
> * Allow all inbound connections.
> 
> So to your point, I was indeed allowing connections to HA, but no luck. Ubiquiti have added some new features for configuring rules so I may take another look in due course. Right now I only have a coupld of Tuya bulbs so I may simply reaplce those in time, but its still annoying not to have solved this.

Did you get this resolved?

**Comment by Nuuki9** [comment]
> Did you get this resolved?

Not really. With the new zone model on UniFi I could have another go, but I only have 1 Tuya bulb now so its not really a priority. I do have it in a segregated VLAN so while it does still have Internet access it at least doesn't have access to the internal network at large. That's good enough, and at some point I'll get around to replacing it.
